lib/pages/job_list_page/job_list_page.dart

```diff
@@ -24,32 +20,26 @@ class JobListPage extends StatefulWidget {
 }
 
 class _JobListState extends State<JobListPage> {
-
   @override
   void initState() {
     super.initState();
   }
 
-  void selectJobCategoryCard(BuildContext ctx) {
-    Navigator.of(ctx).pushNamed(
-      JobSearchPage.routeName,
-      arguments: {
-        'title': 'Job Searching',
-      },
-    );
-  }
 
   @override
   Widget build(BuildContext context) {
-
     return BlocProvider(
-      create: (BuildContext context) => HomeBloc(GraphQLService())..add(FetchHomeData()),
+      create: (BuildContext context) =>
+          HomeBloc(GraphQLService())..add(FetchHomeData()),
       child: Scaffold(
         appBar: AppBar(
           title: Text(widget.title),
           actions: [
             IconButton(
-              onPressed: () {},
+              onPressed: () {
+                Navigator.of(context).push(MaterialPageRoute(
+                    builder: (context) => JobSearchPage(title: 'Searching', searchCategory: '',)));
```

Comment:
suggestion: can make searchCategory as optional field and assign '' by default

Reply:
Applied the suggestion. The field was already optional in fact.